### PR TITLE
Add numeronyms functionality

### LINKED ISSUE #419: Lab 04 - Numeronym
# Lab 4 - Numeronym

- Create an executable go program in directory `04_numeronym/USERNAME`
- Write a function that returns a slice of numeronyms for its input strings:

```
func numeronyms(vals ...string) []string
```

Call `fmt.Println(numeronyms("accessibility", "Kubernetes", "abc"))` in `main` to print:

```
[a11y K8s abc]
```

### PR DESCRIPTION
**Pull request by patrickmarabeas**
Implemented numeronym functionality in a executable Go program which returns a slice of numeronyms given an input of n strings

Fixes #419 

Review of colleague's PR #526

### CODE REVIEW

File: 04_numeronym/patrickmarabeas/main.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	}
+	// Format as "{unicode character}{decimal}{unicode character}"
+	return fmt.Sprintf("%c%d%c", s[0], length-2, s[length-1])
+}
+
+// Convert n strings into a slice of numeronyms
+func numeronyms(vals ...string) []string {
+	result := make([]string, len(vals))
+	i := 0
+	for _, numeronym := range vals {
```
Comment (brvishnu92) [inline 04_numeronym/patrickmarabeas/main.go]: Wouldn't need to declare i separately this way, `for i, numeronym := range vals {`

File: 04_numeronym/patrickmarabeas/main.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// Convert string to a numeronym
+func generateNumeronym(s string) string {
+	length := len(s)
```
Comment (brvishnu92) [inline 04_numeronym/patrickmarabeas/main.go]: length variable creation may be unnecessary as its used in only one case.

So perhaps `if len(s)<4` would suffice.

File: 04_numeronym/patrickmarabeas/main_test.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := fmt.Sprintf("%s\n", "[a11y K8s abc]")
+	got := buf.String()
+
+	t.Run("Main function", func(t *testing.T) {
+		if expected != got {
```
Comment (brvishnu92) [inline 04_numeronym/patrickmarabeas/main_test.go:19]: Consider using testify package  to validate test cases `"github.com/stretchr/testify/assert"`. Its stressed by the team. But, will leave this upto you.

**Review by brvishnu92 — CHANGES_REQUESTED** [review]

File: 04_numeronym/patrickmarabeas/main_test.go (hunk 3 above)
Comment (patrickmarabeas) [reply 04_numeronym/patrickmarabeas/main_test.go:19]: I'll look into it for future labs, thanks.

File: 04_numeronym/patrickmarabeas/main.go (hunk 2 above)
Comment (patrickmarabeas) [reply 04_numeronym/patrickmarabeas/main.go]: It's also used in the return statement twice.

File: 04_numeronym/patrickmarabeas/main.go (hunk 1 above)
Comment (patrickmarabeas) [reply 04_numeronym/patrickmarabeas/main.go]: Good point! Was in the mindset of looping maps : )

**Review by pentaphobe — CHANGES_REQUESTED** [review]
Looking really good, I feel it's ready for approval once there's a colleague PR # referenced

**Comment by patrickmarabeas** [comment]
@pentaphobe PR # reference added.

**Comment by patrickmarabeas** [comment]
Updated doc strings

**Comment by patrickmarabeas** [comment]
Updated to handle multi byte characters

**Review by pentaphobe — APPROVED** [review]

**Comment by pentaphobe** [comment]
LGTM

@brvishnu92 please confirm you're happy with changes, some weren't dismissed automatically since the code lines weren't changed (see final review point)

**Review by brvishnu92 — APPROVED** [review]

**Comment by brvishnu92** [comment]
@patrickmarabeas  LGTM :-)

**Pull request merged**